engine/src/main/java/io/graphqlcrud/types/JdbcTypeMap.java

```diff
@@ -72,38 +66,58 @@ public GraphQLOutputType getAsGraphQLTypeString(int dataType) {
     }
 
     @Override
-    public GraphQLInputType getAsGraphQLFilterType (int dataType) {
-        GraphQLInputType outputType;
+    public GraphQLInputType getAsGraphQLTypeForInput(int dataType, String type) {
```

Comment:
IMO you need to separate concerns of `mutation` and `filter` to different methods and not in single method